### DISCUSSION TITLE
Datetime is not filled correctly

### DISCUSSION THREAD

**Discussion by db1962** [linq2db/linq2db#3156] [Q&A] [answered]
Hi,

## My POCO

```c#
   [Table(Name = "MyTable")]
    public class MyTable
    {
        [PrimaryKey]
        public Guid Id { get; set; }
        public DateTime ScheduledTimeUtc { get; set; }
    }

```

## My Query

```c#
var a =  _connection.MyTable.Where(x=>x.Id == Guid.Parse("7e421f51-072f-40e2-b3f9-000255d4d6b3")).Select(x=>new { x.ScheduledTimeUtc,x.Id}).ToArrayAsync();
a.Wait();
return a.Result;

// result = { ScheduledTimeUtc = {1/01/0001 12:00:00 AM}, Id = {7e421f51-072f-40e2-b3f9-000255d4d6b3} }
```

> Can you please tell me why the application is returning year 1 date and time.
> As you can see using the same query in linqpad with linq2db it is returning valid date.

```c#
var a = MyTable.Where(x => x.Id == Guid.Parse("7e421f51-072f-40e2-b3f9-000255d4d6b3")).Select(x => new { x.ScheduledTimeUtc, x.Id }).ToArrayAsync();
a.Wait();
a.Result.ToJson().Dump();


// result = [{"ScheduledTimeUtc":"2021-07-07T23:04:30","Id":"7e421f51-072f-40e2-b3f9-000255d4d6b3"}]


```

**Comment by MaceWindu** [comment, marked as answer]
`ScheduledTimeUtc` is not marked with `[Column]` attribute so it contains default value. You need to add `[Column]` attribute to it or use `Table("MyTable", IsColumnAttributeRequired = false)` to automatically map all properties to columns

**Reply by jods4** [reply]
Yeah this is a gotcha of linq2db, that is worth keeping in mind:
- If you don't have a `TableAttribute`, then every property is part of the model (unless you use `[NoColumn]`).
- As soon as you put `TableAttribute`, the mode changes to: nothing is part of the model by default, except what is attributed with `[Column]`.
- You can revert to the "everything is in" mode by setting `IsColumnAttributeRequired = false` on `TableAttribute`.

A few times it happened to me as well: everything was working nicely, I wanted to add `[Table]` for some reason and then everything was broken. Never forget `IsColumnAttributeRequired`.

(Of course 0001-01-01 is the `default` value for an uninitialized `DateTime` struct)

**Reply by db1962** [reply]
Thanks for help
